### PR TITLE
Add Belle II tests : Closes #4864

### LINKED ISSUE #4864: Belle II : Add Belle II tests
Motivation
----------
In some recent release, there was a regression affecting RSE expression for RSE following the Belle II convention. In addition, there is no test for Dirac methods.

Modification
------------
Will introduce some Belle II specific tests to cover the issues mentioned above

### PR DESCRIPTION
**Pull request by cserf**
Add Belle II tests : Closes #4864

- Basic tests on B2 schema and add_file method in dirac
- Introduce new `did_name_generator` to generate DIDs valid for multiple VOs
- Fix r`se_name_generator` to match B2 schema

### CODE REVIEW

File: etc/docker/test/matrix_policy_package_tests.yml [hunk 1]
```diff
@@ -1,46 +1,28 @@
-atlas:
+belleii:
```
Comment (maany) [inline etc/docker/test/matrix_policy_package_tests.yml:92]: @bari12 do we want to replace atlas vo-tests with belleii?

File: lib/rucio/common/utils.py [hunk 2]
```diff
@@ -783,7 +783,7 @@ def extract_scope(did, scopes=None, default_extract=_DEFAULT_EXTRACT):
     if not _loaded_policy_package_scope_algorithms:
         register_policy_package_algorithms('scope', _EXTRACT_SCOPE_ALGORITHMS)
         _loaded_policy_package_scope_algorithms = True
-    extract_scope_convention = config_get('common', 'extract_scope', False, None)
+    extract_scope_convention = config_get('common', 'extract_scope', False, None) or config_get('policy', 'extract_scope', False, None)
```
Comment (maany) [inline lib/rucio/common/utils.py:786]: @cserf in this case `extract_scope` in `common` gets a priority over the `extract_scope` in `policy` section. Should it be the other way around?

File: etc/docker/test/matrix_policy_package_tests.yml (hunk 1 above)
Comment (bari12) [reply etc/docker/test/matrix_policy_package_tests.yml:92]: Hmm not really, @cserf what's the motivation here?

File: lib/rucio/common/utils.py (hunk 2 above)
Comment (cserf) [reply lib/rucio/common/utils.py:786]: This is for backward compatibility. I'll remove the common one later

File: etc/docker/test/matrix_policy_package_tests.yml (hunk 1 above)
Comment (cserf) [reply etc/docker/test/matrix_policy_package_tests.yml:92]: I thought that all the autotests were already using ATLAS policy

File: lib/rucio/tests/common.py [hunk 3]
```diff
@@ -64,12 +66,32 @@ def scope_name_generator():
...
+    return 'mock_' + str(uuid())
+
+
 def rse_name_generator(size=10):
     """ Generate random RSE name.
 
     :returns: A random RSE name
     """
-    return 'MOCK_' + ''.join(choice(ascii_uppercase) for x in range(size))
+    return 'MOCK-' + ''.join(choice(ascii_uppercase) for x in range(size))
```
Comment (maany) [inline lib/rucio/tests/common.py:94]: does the '-' work for both atlas and belleii ?

File: tools/bootstrap_tests.py [hunk 4]
```diff
@@ -92,8 +122,11 @@
...
-    for i in additional_test_accounts:
+    for account in additional_test_accounts:
         try:
-            c.add_identity(account='root', identity=i[0], authtype=i[1], email=i[2])
+            client.add_identity(account='root', identity=account[0], authtype=account[1], email=account[2])
         except Exception:
-            print('Already added: ', i)
+            print('Already added: ', account)
+
+    if os.getenv('POLICY') == 'belleii':
```
Comment (maany) [inline tools/bootstrap_tests.py:131]: Minor comment, but just for consistency I think the `config_get` should work as the `rucio.cfg` would already contain the policy package info IMO. Could you check @cserf. Thanks

File: lib/rucio/tests/common.py (hunk 3 above)
Comment (cserf) [reply lib/rucio/tests/common.py:94]: Yes

File: tools/bootstrap_tests.py (hunk 4 above)
Comment (cserf) [reply tools/bootstrap_tests.py:131]: Will check

Comment (cserf) [reply tools/bootstrap_tests.py:131]: I checked with the config_get but it fails

Comment (maany) [reply tools/bootstrap_tests.py:131]: Okie thanks @cserf , I will add it to my list of todo and modify this code later on :)

**Review by maany — APPROVED** [review]
Looks good. Thanks @cserf

**Pull request merged**